### ISSUE TITLE
[no-unsafe-member-access] Namespaces trigger error

### ISSUE THREAD

**Issue by mf-pherlihy** [typescript-eslint/typescript-eslint#1868]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-member-access": 2
  }
}
```

```TS
// some-definition.d.ts
declare class SomeClass extends NodeJS.EventEmitter {
 //
}
```

**Expected Result**

No error

**Actual Result**

```
eslint: error
@typescript-eslint/no-unsafe-member-access - Unsafe member access .EventEmitter on an any value
```

**Additional Info**

Pretty simple to reproduce, just create a definition file and have a class that extends the `NodeJS.EventEmitter` class.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.27.0` |
| `@typescript-eslint/parser`        | `2.27.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.18.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
Could you please create a repro repo for this?
I am unable to repro this against master.

I suspect that there is an issue with your config.

**Comment by mf-pherlihy** [comment]
@bradzacher Please see: https://github.com/mf-pherlihy/typescript-eslint-no-unsafe-member-access

**Comment by bradzacher** [comment]
interesting. It looks like it's an error due to v`13.11.1`  of `@types/node`.
I can't repro it against master, because we have `12.12.7` installed, which is working fine.

I'm not sure what is happening.

**Comment by bradzacher** [comment]
doing a pseudo-bisect shows that this is broken from v`13.1.0` onwards of `@types/node`, with v`12.12.35` (the latest v12 release at the time) working fine.

**Comment by bradzacher** [comment]
I have a repro without any of our packages, so I'm not sure what the problem is. I'll need to keep looking
https://github.com/bradzacher/types-node-any-repro

**Comment by bradzacher** [comment]
Looks like you stumbled upon a bug in typescript!

https://github.com/microsoft/TypeScript/issues/37901

**Comment by tao-cumplido** [comment]
I can still reproduce this with 

- eslint 7.12.1
- @typescript-eslint/eslint-plugin 4.6.0
- typescript 4.0.5

the linked issue is supposedly fixed since typescript 4.0.1

```ts
namespace A {
	export interface B {}
}

export interface C extends A.B {} // Unsafe member access .B on an any value. eslint(@typescript-eslint/no-unsafe-member-access)
```
